### PR TITLE
Add a spell checker for markdown.

### PR DESCRIPTION
**Pull request by mreid-moz**
Addresses #42

### CODE REVIEW

File: cookbooks/hll_zeppelin.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ spark.udf.register("hll_cardinality", HllCardinality)
 
 ## Example Usage
 
-This is a short example which can also be used to verify expected behavior.
+This is a short example which can also be used to verify expected behaviour.
```
Comment (harterrt) [inline cookbooks/hll_zeppelin.md]: Let's not favo[u]r the British spelling.

File: datasets/batch_view/retention/intro.md [hunk 2]
```diff
@@ -14,17 +14,17 @@ pass while avoiding the problem of double counting. This data-structure has the
...
-operation. With SQL primitive, this requires a recomputation of COUNT DISTINCT
+operation. With SQL primitive, this requires a recalculation of COUNT DISTINCT
 over `client_id`'s in the 7-day window.
 
 #### Background and Caveats
 
 1. The data starts at 2017-03-06, the [merge date where Nightly started to
    track Firefox 55 in Mozilla-Central][release_calendar]. However, there was
-not a consistent view into the behavior of first session profiles until the
+not a consistent view into the behaviour of first session profiles until the
```
Comment (harterrt) [inline datasets/batch_view/retention/intro.md]: Please add exception for "behavior".

File: tools/spark.md [hunk 3]
```diff
@@ -211,23 +213,22 @@ See also the
...
+see [Choosing a Dataset](/concepts/choosing_a_dataset.md).
 
 These datasets are optimized for fast access, and will far out-perform
-analysis on the raw telemetry ping data.
+analysis on the raw Telemetry ping data.
 
 ### Accessing the Spark UI
 
-Go to https://localhost:8888/spark after ssh-ing into the spark cluster
+After establishing an SSH connection to the Spark cluster, go to https://localhost:8888/spark
```
Comment (harterrt) [inline tools/spark.md]: What a great revision. I didn't realize spell-checking would make us limit our use of jargon.

File: tools/spark.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ To schedule a Spark job:
 3.  Enter some details:
     1.  The “Job Name” field should be a short descriptive name, like
         “chromehangs analysis”.
-    2.  Upload your IPython notebook containing the analysis.
+    2.  Upload your Jupyter notebook containing the analysis.
```
Comment (harterrt) [inline tools/spark.md]: Another unexpected benefit! Standardizing terminology!

File: cookbooks/hll_zeppelin.md (hunk 1 above)
Comment (mreid-moz) [reply cookbooks/hll_zeppelin.md]: You mean Canadian? :)

`markdown-spellcheck` comes with a US dictionary as well, but it seems to be missing a lot more sort-of-jargon words so I went with the default (which was, surprisingly, *not* the US version).

I will change it in favor the US dictionary.

Comment (harterrt) [reply cookbooks/hll_zeppelin.md]: Does the US dictionary allow both spellings or just the US spelling? I'd suggest allowing both.

Comment (mreid-moz) [reply cookbooks/hll_zeppelin.md]: The US dictionary is just as picky about the superfl[u]o[u]s u's.

The spelling library's docs say:
> We default to British English but will change in the next major to American.

So I think we're better off and more future-proof by using the US dictionary.  I've updated the spellings accordingly.

**Comment by mreid-moz** [comment]
@harterrt I've added a section about spelling to the Contributing article, RFAL.

**Pull request merged**